staging/src/k8s.io/apiserver/pkg/util/webhook/webhook.go

```diff
@@ -112,6 +113,11 @@ func WithExponentialBackoff(initialBackoff time.Duration, webhookFn func() error
 		if _, shouldRetry := apierrors.SuggestsClientDelay(err); shouldRetry {
 			return false, nil
 		}
+		// "Connection reset by peer" is usually a transient error, but it's not
+		// retried by the rest client because the request is not GET.
+		if net.IsConnectionReset(err) {
```

Comment:
shouldn't this move up to the first if-clause?

Reply:
I would move IsConnectionReset and also IsInternalError up. Both are unconditional, as the others. Then for all other errors we call `SuggestsClientDelay`. Now in the current shape it looks like the order of the tests matter. But it doesn't.